firefox_ui_tests/functional/security/test_grey_larry.py

```diff
@@ -0,0 +1,75 @@
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver.errors import NoSuchElementException
+
+from firefox_ui_harness.decorators import skip_if_e10s, skip_under_xvfb
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestGreyLarry(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.url = self.marionette.absolute_url('layout/mozilla.html')
+
+        self.identity_popup = self.browser.navbar.locationbar.identity_popup
+
+    def tearDown(self):
+        try:
+            self.identity_popup.close(force=True)
+            self.windows.close_all([self.browser])
+        except NoSuchElementException:
+            # TODO: A NoSuchElementException may be thrown here when the test is skipped
+            # as under xvfb.
+            pass
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_if_e10s
+    @skip_under_xvfb
+    def test_grey_larry(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.url)
+
+        # Check the favicon
+        # TODO: find a better way to check, e.g., mozmill's isDisplayed
+        favicon = self.browser.navbar.locationbar.favicon
+        self.wait_for_condition(lambda _: favicon.get_attribute('hidden') == 'false')
+
+        # Check that the identity popup organization label is blank
+        self.assertEqual(self.identity_popup.organization_label.get_attribute('value'),
+                         '')
+
+        self.identity_popup.box.click()
+        self.wait_for_condition(lambda _: self.identity_popup.is_open)
+
+        # Check the idenity popup doorhanger
+        self.assertEqual(self.identity_popup.popup.get_attribute('className'),
+                         'unknownIdentity')
+
+        # Open the Page Info window by clicking the More Information button
+        page_info = self.browser.open_page_info_window(
+            lambda _: self.identity_popup.more_info_button.click())
+
+        try:
+            # Verify that the current panel is the security panel
+            self.assertEqual(page_info.deck.selected_panel, page_info.deck.security)
+
+            # Check for localhost in the domain listed on the security panel
+            self.assertIn('127.0.0.1',
+                          page_info.deck.security.domain.get_attribute('value'))
+
+            # Check the owner label for localized 'securityNoOwner'
+            self.assertEqual(page_info.deck.security.owner.get_attribute('value'),
+                             self.browser.get_property('securityNoOwner'))
+
+            # Check the verifier label for localized 'notset'
+            self.assertEqual(page_info.deck.security.verifier.get_attribute('value'),
+                             self.browser.get_property('notset'))
+
+            page_info.close()
+        finally:
+            self.browser.focus()
```

Comment:
If some test fails for page info the window will not be closed. So move line 73 right before that one.